### PR TITLE
Reformat all code base with Standard Style Guide

### PR DESCRIPTION
**Pull request by pftg**

### CODE REVIEW

**Comment by pftg** [comment]
@jcupitt and here is the epic changes, please check

File: example/draw_lines.rb [hunk 1]
```diff
@@ -18,11 +18,11 @@
 starting = Process.clock_gettime(Process::CLOCK_MONOTONIC)
 
 lines = image
-lines = lines.mutate do |x|
-  (0 .. 1).step 0.01 do |i|
+lines = lines.mutate { |x|
+  (0..1).step 0.01 do |i|
```
Comment (jcupitt) [inline example/draw_lines.rb:22]: This one feels odd to me. Multiline `{ }` look ugly, and it's not a functional block.

File: example/example2.rb [hunk 2]
```diff
@@ -20,15 +20,15 @@
   puts ""
   puts "call #{i} ..."
   out = Vips::Operation.call "black", [200, 300]
-  if out.width != 200 or out.height != 300
+  if (out.width != 200) || (out.height != 300)
```
Comment (jcupitt) [inline example/example2.rb:23]: I'd remove the brackets, I've not used them anywhere else.

File: lib/vips/image.rb [hunk 3]
```diff
@@ -25,35 +25,35 @@ module Vips
   attach_function :vips_foreign_find_load_buffer, [:pointer, :size_t], :string
   attach_function :vips_foreign_find_save_buffer, [:string], :string
 
-  if Vips::at_least_libvips?(8, 9)
+  if Vips.at_least_libvips?(8, 9)
```
Comment (jcupitt) [inline lib/vips/image.rb:28]: I don't really like this rule either --- it's `Vips::Image.new_from_file`, but `Vips.at_least?`. This feels inconsistent.

It ought to always be Module::method, I think. But perhaps we just have to live with adding some ugliness.

File: lib/vips/image.rb [hunk 4]
```diff
@@ -486,13 +486,11 @@ def self.new_from_array array, scale = 1, offset = 0
       image = Vips::Image.matrix_from_array width, height, array
       raise Vips::Error if image.nil?
 
-      image = image.mutate do |mutable|
+      image.mutate { |mutable|
```
Comment (jcupitt) [inline lib/vips/image.rb:489]: I think this should be `image = image.mutate { |mutable|`

Comment (jcupitt) [reply lib/vips/image.rb:489]: Ah wait, you're removing the return, OK.

File: lib/vips/image.rb [hunk 5]
```diff
@@ -1276,17 +1272,17 @@ def maxpos
...
+      [v, x, y]
     end
 
     # a median filter
     #
     # @param size [Integer] size of filter window
     # @return [Image] result of median filter
     def median size = 3
-      rank size, size, (size * size) / 2
+      rank size, size, (size**2) / 2
```
Comment (jcupitt) [inline lib/vips/image.rb:1285]: You could remove the brackets here.

File: lib/vips/methods.rb [hunk 6]
```diff
@@ -1,2771 +1,2769 @@
...
-
-# @!method globalbalance(**opts)
-#   Global balance an image mosaic.
-#   @param opts [Hash] Set of options
-#   @option opts [Float] :gamma Image gamma
-#   @option opts [Boolean] :int_output Integer output
-#   @return [Vips::Image] Output image
-
+    # @!method self.system(cmd_format, **opts)
+    #   Run an external command.
```
Comment (jcupitt) [inline lib/vips/methods.rb:4]: This file is generated automatically, you'll need to change the generator rather than editing it.

**Comment by jcupitt** [comment]
I saw a few small things, otherwise it looks great! Nice work!

Your fingers must be aching.

File: example/draw_lines.rb (hunk 1 above)
Comment (pftg) [reply example/draw_lines.rb:22]: yep, this somehow become popular to use `{}` for blocks with returns

File: lib/vips/methods.rb (hunk 6 above)
Comment (pftg) [reply lib/vips/methods.rb:4]: I see, let's skip it from linting

File: lib/vips/image.rb (hunk 3 above)
Comment (pftg) [reply lib/vips/image.rb:28]: `Module::method` it's very uncommon. At least on the projects where I have been.

**Comment by pftg** [comment]
@jcupitt I'll check if it's possible to disable semantics blocks, but `||` and `Module.method` will be great to have, it's wide using. As I got you are trying to avoid parentheses, but in several places better to use them then uncommon paractices.

File: lib/vips/methods.rb (hunk 6 above)
Comment (pftg) [reply lib/vips/methods.rb:4]: Will skip it from linter.

File: example/draw_lines.rb (hunk 1 above)
Comment (pftg) [reply example/draw_lines.rb:22]: not popular, but has been preselected by authors of the standard. They prefer to have semantics blocks

Comment (pftg) [reply example/draw_lines.rb:22]: disabled semantic blocks

File: lib/vips/image.rb (hunk 5 above)
Comment (pftg) [reply lib/vips/image.rb]: done

File: lib/vips/methods.rb (hunk 6 above)
Comment (pftg) [reply lib/vips/methods.rb]: done

File: lib/vips/image.rb (hunk 3 above)
Comment (jcupitt) [reply lib/vips/image.rb:28]: Hmmm, but `Module::method` is just using `::` for namespace resolution, isn't it? It's not a method call. It's exactly like `Module::Class`. Or does the `::` refer to the thing to the *right* of the operator? That seems weird.

I think it's ugly and illogical, but if it's common practice, I can see the advantage.

File: example/example2.rb (hunk 2 above)
Comment (jcupitt) [reply example/example2.rb:23]: To explain: the whole of ruby-vips does `xxx || yyy`, so if we add brackets here, we should add them everywhere.

Personally, I don't like brackets in chained `||` and `&&` expressions, they add noise and reduce clarity (imo). If you have an expression that mixes `&& and `||` then you need them, but that types of conditional is rare (and probably too complicated anyway).

I think I would change this to:

```ruby
  if out.width != 200 || out.height != 300
```

Comment (pftg) [reply example/example2.rb]: @jcupitt updated

**Comment by jcupitt** [comment]
Fantastic! Nice work again.

**Pull request merged**